Fixes related output support for fflive

pawelzwronek opened the pull request:
Add support for pipe('-') as output
Fix missing header file for memmove func

pawelzwronek committed:
Add support for pipe('-') as output
Fix missing header file for memmove func

ramiropolla commented:
Hi,

I merged two chunks (missing header and changing `-` to `pipe:`) in separate commits. What is your use-case for using fflive with output to pipe? There's nothing wrong with it, but I'm curious :)

Regarding the warning message and `exit(1)`, I think it's ok to have output without script (which would be the same as replication in ffedit).

By the way, your git configuration doesn't seem correct (username and email). Do you want to fix the commit with proper credentials? (I don't mind force pushing the last few commits to fix this).

Ramiro

pawelzwronek committed:
fflive: fix 'process crashed' on macOS

pawelzwronek committed:
Fix building as shared libraries

ramiropolla commented:
I won't have time to properly test all of these changes for a few weeks. But I have a few comments:
- Can you tell me how to reproduce the crash on macos?
- Instead of using signals, perhaps you can implement scripting support to control the playback for fflive.
- JS_FreeValue* should stay in the header file (otherwise it can slow things down)
- Remove the cosmetic (whitespace) changes in your patches. I don't like the inconsistent whitespace from quickjs either, but I'm trying to keep it as close as possible to the upstream repository (IIRC they fixed all their whitespace issues some time ago)
- I wanted to reduce the usage of the global ffe_class (preferably remove it). Can you try using local contexts (outside of fftools/ffedit*.c) instead?
- Why the wildcard change in library.mak?

Also could you please try to split the patch in one commit for each functional change?

Thanks!

pawelzwronek commented:
My bad. These commits was meant for my build for LiveMosher only, I'll fix that.

- When you run `ffgac -nostats -i 2.mp4 -an -mpv_flags +nopimb+forcemv -qscale:v 0 -g max -sc_threshold max -vcodec mpeg4 -vf scale=1280:720 -f rawvideo - | fflive -i - -s ../scripts/script1.js  -o - | ffgac -i - -i 2.mp4 -map 0:v -map 1:a -c:a copy -c:v libx264 -preset medium 4.mp4` and close fflive window in the middle I get 
![obraz](https://github.com/user-attachments/assets/feb7cd68-ea7c-4d57-a016-9a058cd05394)
- What kind of scripting do you think of? I was thinking about `zmq` at the beginning but ended with signals as easier(faster to code) method.
- It crossed my mind but as I said it's meant for my build only :)
- Change in `library.mak` fixed missing some `*.lib` at the end of the build on Windows. It skip the operation when the file is not found

I'll undo those commits. I should have make this PB in a new branch. How about to close it after my fixes?

pawelzwronek commented:
You can find those changes in `https://github.com/pawelzwronek/ffglitch-core/tree/LiveMosher`

pawelzwronek commented:
>     * Instead of using signals, perhaps you can implement scripting support to control the playback for fflive.
In my LiveMosher branch I removed signal based control and replaced it with zmq simple command over ipc or tcp.
>     * JS_FreeValue* should stay in the header file (otherwise it can slow things down)
Done. Not so clean solution, but I had no other idea.
>     * Remove the cosmetic (whitespace) changes in your patches. I don't like the inconsistent whitespace from quickjs either, but I'm trying to keep it as close as possible to the upstream repository (IIRC they fixed all their whitespace issues some time ago)
Done.
>     * I wanted to reduce the usage of the global ffe_class (preferably remove it). Can you try using local contexts (outside of fftools/ffedit*.c) instead?
Done.
>     * Why the wildcard change in library.mak?
Done.
